sympy/functions/combinatorial/factorials.py

```diff
@@ -848,15 +854,22 @@ def eval(cls, n, k):
         d = n - k
         if d.is_zero or k.is_zero:
             return S.One
-        elif d.is_zero is False:
-            if (k - 1).is_zero:
-                return n
-            elif k.is_negative:
+        if (k - 1).is_zero:
+            return n
+        if k.is_integer:
+            if k.is_negative:
                 return S.Zero
-            elif n.is_integer and n.is_nonnegative and d.is_negative:
+            if n.is_integer and n.is_nonnegative and d.is_negative:
                 return S.Zero
-        if k.is_Integer and k > 0 and n.is_Number:
-            return cls._eval(n, k)
+            if n.is_Number:
+                return cls._eval(n, k)
+        elif n.is_negative and n.is_integer and not k.is_integer:
+            # a special case when binomial evaluates to complex infinity
+            return S.ComplexInfinity
+        elif k.is_number:
```

Comment:
@jksuom will it be a good idea to use `is_number`, because `k.is_Number` won't all complex and imaginary numbers.

Reply:
It looks like `is_number` could be better than `is_Number`.